lib/ex_webrtc/media/ivf_writer.ex

```diff
@@ -0,0 +1,88 @@
+defmodule ExWebRTC.Media.IVFWriter do
+  @moduledoc """
+  Writes video frames as an IVF file.
+  """
+
+  alias ExWebRTC.Media.IVFFrame
+
+  @opaque t() :: %__MODULE__{
+            file: File.io_device(),
+            frames_cnt: non_neg_integer()
+          }
+
+  @enforce_keys [:file]
+  defstruct @enforce_keys ++ [update_header_after: 0, frames_cnt: 0]
+
+  defguard update_header?(writer)
+           when writer.frames_cnt >= writer.update_header_after and
+                  rem(writer.frames_cnt, writer.update_header_after) == 0
+
+  @doc """
+  Creates a new IVF writer.
+
+  Initially, IVF header is written with `num_frames` and 
+  is updated every `num_frames` by `num_frames`.
```

Comment:
I wonder if it wouldn't be better to explicitly update the header after all the frames are written? Updating every x frames seems kind error-prone, what if no frames are available, but the header was not updated? Updating header every new frame also seems wastefuf.

I'm not familiar with IVF, but that's just  an observation.

Reply:
I can add `close` function that I had before that will update header when called. However, it looks like there are some cases where we are not able to call `close` e.g. sending SIGINT (CTRL+C) when executing Elixir script.

I think this should be okay to have a header indicating more frames than there are in the file. Also, Pion starts with I 900 frames and updates header on calling `close`.